portfolio/src/main/webapp/script.js

```diff
@@ -17,12 +17,73 @@
  */
 function addRandomFact() {
   const facts =
-      ['I have a pet dog', 'I like to code'];
+      ['I have a pet dog!', 'I like to code!', 'Marquees are my favourite HTML element!'];
 
   // Pick a random fact.
   const fact = facts[Math.floor(Math.random() * facts.length)];
 
   // Add it to the page.
   const factContainer = document.getElementById('fact-container');
   factContainer.innerText = fact;
-}
\ No newline at end of file
+}
+
+/**
+ * Controls slideshow image display on the page
+ */
+const deckNums = { // Enumerate slide decks
+  DOG_DECK: 0,
+  PLACE_DECK: 1,
+  FOOD_DECK: 2,
+};
+
+const PREV_SLIDE_DIR = -1;
+const NEXT_SLIDE_DIR = 1;
+
+class Slideshow {
+  constructor(idNum, slides, slideshowDOM) {
+    this.index = 0;
+    this.idNum = idNum;
+    this.slides = slides;
+    this.slideshowDOM = slideshowDOM;
+    this.prevButton = this.slideshowDOM.querySelectorAll(".prev-slide-button")[0];
+    this.nextButton = this.slideshowDOM.querySelectorAll(".next-slide-button")[0];
+    this.addButtons();
+    this.preloadImages();
+  }
+  changeSlide(direction) {
+    const deckSize = this.slides.length;
+    this.index += direction;
+    this.index = (this.index % deckSize + deckSize) % deckSize; // wrap around
+    let imageDOM = this.slideshowDOM.querySelectorAll("img")[0];
```

Comment:
Instead of querySelectorAll()[0] you can use querySelector, which returns the first match. https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector